drivers/SmartThings/harman-luxury/src/devices.lua

```diff
@@ -0,0 +1,142 @@
+local log = require "log"
+local st_utils = require "st.utils"
+
+local api = require "api.apis"
+
+Devices = {}
+
+local suppourted_devices = { "L75ms", "L42ms", "AVR5" }
+
+local MNID = "0BE8"
+local DEFAULT_DEVICE_NAME = "HarmanLuxury"
+local DEFAULT_MANUFACTURE_NAME = "Harman Luxury Audio"
+local DEFAULT_MODEL_NAME = "Harman Luxury"
+local DEFAULT_PRODUCT_NAME = "Harman Luxury"
+
+local devices_SetupID = {
+    L75ms = "602",
+    L42ms = "603",
+    AVR5 = "604"
+}
+
+function Devices.GetSupportedDevices()
+    log.info("GetSupportedDevices: supported models: " .. st_utils.stringify_table(suppourted_devices))
+    return suppourted_devices
+end
+
+local function GetDefaultDeviceInfo(dni, ip)
```

Comment:
Since we want the values here to not only be non-nil, but also explicitly `strings`, it would be more defensive and less bug-prone to use `if type(var) ~= "string"` instead of `if not var`.

So for example, `if type(label) ~= "string"` instead of `if not label`. This check will catch both `nil` and cases where the value passed in is not a string.